docs/proposals/202005_version_documentation.md

```diff
@@ -0,0 +1,164 @@
+---
+title: 'Building A Versioning Plugin For Thanos'
+type: proposal
+menu: proposals
+status: approved
+owner: thisisobate
+Date: May 2020
+---
+
+
+## Problem
+
+The Thanos website contains the docs area which is built by rendering the markdown files from the tip of master of the Thanos repository. This frequently causes confusion for users, as different Thanos versions have different features.
+
+In this proposal, we want to solve this. In particular, we want to:
+
+1. Build a Version Picker that serves as a tool which will aid easy access to other versions of the documentation. This picker will Include links to change version (the version must be in the URL).
+2. A solid and robust Documentation structure.
+3. Design a workflow for managing docs that integrates with Thanos Git workflow ie updating corresponding docs on pull requests, cherry-picks, etc.
+
+## Motivation
+
+Many Users (mostly developers) would often want to look through the docs of previous releases, but searching for it manually by looking through the entire GitHub repository is time-consuming and ineffective. Because the site is built by rendering the markdown files from the tip of master of the Thanos repository, it causes confusion for users, as breaking changes are often merged into master. The versioning plugin allows the user to access the docs of both latest and previous releases on the Thanos documentation page, and it also allows the developer to fetch, write, and fix the docs of both latest and previous releases.
+
+
+## Requirements
+
+#### User Story (Latest)
+
+* As a Thanos developer, I want to be able to just write docs for my current `master` version.
+* As a Thanos developer, I want to build website within versioned docs with a single action.
```

Comment:
what do we mean by `within`? Should it be `with`?

Reply:
exactly...`with` is the correct word to use.
Thanks for pointing this out :)